lib/compress.js

```diff
@@ -2893,7 +2920,7 @@ merge(Compressor.prototype, {
                 });
             }
         }
-        return self;
+        return self.evaluate(compressor)[0];
```

Comment:
@kzc correct me if I'm wrong, but without `unsafe` any `AST_PropAccess._eval(compressor)` will just hit `throw def;` straight away, so `AST_Sub.evaluate(compressor)` should always be a no-op when `unsafe` is disabled.

Before this change, `AST_Sub.evaluate(compressor)` is always a no-op _regardless_ of `unsafe`, so calling it or not makes no difference, apart from may be better performance.